5. 

The first program would output the integer 3. 
The second program would give an error, as calling the 'x' variable outside of the 3.times block is outside scope. Inner scope can access outside, but not vice versa.